operate/client/src/modules/queries/elementInstances/useElementInstance.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+
+import {skipToken, useQuery, UseQueryResult} from '@tanstack/react-query';
+import {fetchElementInstance} from '../../api/v2/elementInstances/fetchElementInstance';
+import {ElementInstance} from '@vzeta/camunda-api-zod-schemas';
+import {RequestError} from '../../request';
+
+const ELEMENT_INSTANCE_QUERY_KEY = 'elementInstance';
+
+function getQueryKey(elementInstanceKey?: string) {
+  return [ELEMENT_INSTANCE_QUERY_KEY, elementInstanceKey];
+}
+
+const useElementInstance = <T = ElementInstance>(
+  elementInstanceKey: string,
+  select?: (data: ElementInstance) => T,
+): UseQueryResult<T, RequestError> => {
+  return useQuery({
+    queryKey: getQueryKey(elementInstanceKey),
+    queryFn: elementInstanceKey
+      ? async () => {
+          const {response, error} = await fetchElementInstance({
+            elementInstanceKey,
+          });
+
+          if (response !== null) return response;
+          throw error;
+        }
```

Comment:
I know this is typed on the other hooks but it was a mistake doing that. The library maintainers recommend the type inference work instead of explicitly typing it. I intend to fix the other existing hooks in an extra PR but for now we can simply do the following:

```suggestion
) => {
  return useQuery({
    queryKey: getQueryKey(elementInstanceKey),
    queryFn: elementInstanceKey
      ? async () => {
          const {response, error} = await fetchElementInstance({
            elementInstanceKey,
          });

          if (response !== null) return response as Promise<ElementInstance>;
          throw error;
        }
```